packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -1472,6 +1627,121 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
+  /// or trailing scrollbar's buttons.
+  @protected
+  @mustCallSuper
+  void handleButtonsPressDown(
+    Offset localPosition,
+    bool isLeadingButton, {
+      Duration duration = const Duration(milliseconds: 100),
+    }
+  ) {
+    assert(_debugCheckHasValidScrollPosition());
+    _currentController = widget.controller ?? PrimaryScrollController.of(context);
+
+    if (isLeadingButton) {
+      setState(() { _leadingButtonPressed = true; });
+    } else {
+      setState(() { _trailingButtonPressed = true; });
+    }
+
+    double scrollIncrement;
+    // Is an increment calculator available?
+    final ScrollIncrementCalculator? calculator = Scrollable.of(
+      _currentController!.position.context.notificationContext!,
+    )?.widget.incrementCalculator;
+
+    if (calculator != null) {
+      scrollIncrement = calculator(
+        ScrollIncrementDetails(
+          type: ScrollIncrementType.line,
+          metrics: _currentController!.position,
+        ),
+      );
+    } else {
+      // Default line increment
+      scrollIncrement = 50.0;
+    }
+
+    if (isLeadingButton) {
+      // Adjust scrollIncrement for direction
+      switch (_currentController!.position.axisDirection) {
+        case AxisDirection.right:
+        case AxisDirection.down:
+          scrollIncrement = math.min(scrollIncrement, _currentController!.position.extentBefore);
+          scrollIncrement = -scrollIncrement;
+          break;
+        case AxisDirection.up:
+        case AxisDirection.left:
+          scrollIncrement = math.min(scrollIncrement, _currentController!.position.extentAfter);
+          break;
+      }
+    } else {
+      // Adjust scrollIncrement for direction
+      switch (_currentController!.position.axisDirection) {
+        case AxisDirection.right:
+        case AxisDirection.down:
+          scrollIncrement = math.min(scrollIncrement, _currentController!.position.extentAfter);
+          break;
+        case AxisDirection.up:
+        case AxisDirection.left:
+          scrollIncrement = math.min(scrollIncrement, _currentController!.position.extentBefore);
+          scrollIncrement = -scrollIncrement;
+          break;
+      }
+    }
+
+    if (scrollIncrement == 0.0) {
+      return;
+    }
+
+    _currentController!.position.moveTo(
+      _currentController!.position.pixels + scrollIncrement,
+      duration: duration,
+      curve: Curves.linear,
+    );
+  }
+
+  /// Handler called when the pointer stops contacting the buttons.
+  @protected
+  @mustCallSuper
+  void handleButtonsPressUp(bool isLeadingButton) {
+    if (isLeadingButton) {
+      setState(() { _leadingButtonPressed = false; });
+    } else {
+      setState(() { _trailingButtonPressed = false; });
+    }
+  }
+
+  /// Handler called when a press on the leading or trailing scrollbar's buttons
+  /// has been recognized.
+  @protected
+  @mustCallSuper
+  void handleButtonsLongPress(bool isLeadingButton) {
```

Comment:
For all of these with the `isLeadingButton` bool methods, making this a required named parameter will help when reading where they are being called.